src/Nethermind/Nethermind.Merge.Plugin/Handlers/GetPayloadHandlerBase.cs

```diff
@@ -37,11 +41,19 @@ protected GetPayloadHandlerBase(int apiVersion, IPayloadPreparationService paylo
             return ResultWrapper<TGetPayloadResult?>.Fail("unknown payload", MergeErrorCodes.UnknownPayload);
         }
 
+        TGetPayloadResult getPayloadResult = GetPayloadResultFromBlock(blockContext);
+
+        if (getPayloadResult is GetPayloadV2Result getPayloadWrapperResult && !getPayloadWrapperResult.ExecutionPayload.IsProperFork(_specProvider))
```

Comment:
Looks very bad. Polymorphic call is the way.

Reply:
payloadresultv1 is ExecutionPayload itself, so we need t check the result is at least V2, that's why it's so ugly looking. In overall adding new features to jsonrpc is hard. Let me try to improve this